includes/sanitizers/class-amp-meta-sanitizer.php

```diff
@@ -133,11 +133,81 @@ protected function ensure_charset_is_present() {
 	 * Always ensure we have a viewport tag.
 	 *
 	 * The viewport defaults to 'width=device-width', which is the bare minimum that AMP requires.
+	 * If there are @viewport style rules, these will have been moved into the content attribute of their own meta[name="viewport"] tags.
+	 * So this iterates over all of those tags, gets the content value,
+	 * and merges all of the valid values into a single meta tag.
 	 */
 	protected function ensure_viewport_is_present() {
 		if ( empty( $this->meta_tags[ self::TAG_VIEWPORT ] ) ) {
 			$this->meta_tags[ self::TAG_VIEWPORT ][] = $this->create_viewport_element( static::AMP_VIEWPORT );
+		} else {
+			$parsed_rules = [];
+
+			// Reverse the meta[name="viewport"] tags, so original meta[name="viewport"] tags have precedence over @viewport style rules.
+			foreach ( array_reverse( $this->meta_tags[ self::TAG_VIEWPORT ] ) as $meta_viewport ) {
+				$viewport_content = explode( ',', $meta_viewport->getAttribute( 'content' ) );
+				foreach ( $viewport_content as $rule ) {
+					$exploded_rule = explode( '=', $rule, 2 );
+					if ( ! isset( $exploded_rule[1] ) ) {
+						continue;
+					}
+
+					list( $name, $value )          = $exploded_rule;
+					$parsed_rules[ trim( $name ) ] = trim( $value );
+				}
+			}
+
+			$valid_rules                           = $this->get_valid_viewport_rules( $parsed_rules );
```

Comment:
Yeah, they are removed 😄 

Though it might be confusing how the validation error references '<meta>`.

For example, on adding this to a Custom HTML block:

```html
<style>@viewport { disallowed: 100px }</style>
```

...here's the validation error:

![validation-error-here](https://user-images.githubusercontent.com/4063887/76660553-cc688100-653e-11ea-8eac-6dedf3945477.png)

Reply:
Yeah, but better than silently removing them. Later we can think about adding attribution.